docs/user/developing_jobs.md

```diff
@@ -42,21 +44,125 @@ In brief, the following general steps can be followed:
+        if child_region_created:
+            self.add(child_region)
+            parent_region.add_child(child_region)
+        site, site_created = self.get_or_instantiate("site", ids={"name": location.site.name})
+        if site_created:
+            self.add(site)
+            child_region.add_child(site)
+        location, location_created = self.get_or_instantiate("location", ids={"name": location.name})
+        if location_created:
+            self.add(location)
+            site.add_child(location)
+```
+
+As an additional bonus, this way the code has fewer levels of indentation.
+
+The essence of this is that you should make liberal use of `select_related` to join together the database tables you need into a single, big query rather than a bunch of small queries.
 
-The built-in implementation of `sync_data` is composed by 4 steps:
+!!! warning
+    If you are using Nautobot 1.4 or lower, `CACHEOPS_ENABLED` is set to `True` by default. As long as this is the case, you should use `select_related`'s cousin `prefetch_related` instead (see [here](https://github.com/Suor/django-cacheops#caveats) for cacheops documentation on that matter). In 1.5 this is disabled by default, but if you explicitly turn it on you will again need to use `prefetch_related` instead.
+    
+    If you are using Nautobot 2.0 or higher, this warning can be ignored as cacheops has been removed from Nautobot. 
+
+!!! note
+    Check out the [Django documentation](https://docs.djangoproject.com/en/3.2/topics/db/optimization/) for a more comprehensive source on optimizing database access.
+
+### Optimizing worker stdout IO
+
+If after optimizing your database access you are still facing performance issues, you should check out the [analyzing job performance](#analyzing-job-performance) section of the docs. Should you find that a certain `io.write` appears high up in the ranking, you are probably facing an issue where your job is writing to stdout so quickly that your worker node/process cannot drain its buffer quickly enough. To deal with this, tone down on what you are logging to stdout inside your job. This could be any of the following things (non-exhaustive, check out your worker logs):
+
+- diffsync [logging configuration](https://diffsync.readthedocs.io/en/latest/api/diffsync.logging.html?highlight=logging)
+- `print` calls
+- other, external frameworks you are using
+
+### Minimizing external IO
+
+In most if not all cases, the side of an SSoT job that interacts with the non-Nautobot system will be accessed through some form of IO as for example HTTP requests via the network. Depending on the amount of requests, request/response size and the latency to the remote system this can take a lot of time. Care should be taken when crafting the IO interaction, using bulk endpoints instead of querying each individual record on the remote system where possible.
+
```

Comment:
```suggestion

As an example, you could take the following proposal:

* Collect Sites
    * For each Site, collect all Devices
        * For each Device collect the Interface information
        * For each Device collect the VLAN information

However, this would be preferred:

* Collect all Sites
* Collect all Devices
* Collect all Interfaces
* Collect all VLANs

```

Reply:
I worked this in, thanks!